### PR TITLE
Fortinet's FortiOS web proxy profile

### PR DESCRIPTION
**Pull request by migumun**
##### SUMMARY

Fortinet is adding Ansible support for FortiOS and FortiGate products. This module follows the same structure, guidelines and ideas given in previous approved module for a parallel feature of FortiGate (webfiltering): https://github.com/ansible/ansible/pull/37196 
In this case we are providing a different functionality: "Web Proxy Profile".

Please note that this will be part of other modules to come for FortiGate, including different functionalities: system, wireless-controller, firewall, webfilter, ips, web-proxy, wanopt, application, dlp spamfilter, log, vpn, certificate, user, dnsfilter, antivirus, report, waf, authentication, switch controller, endpoint-control and router. We plan to follow the same style, structure and usage as in the previous module in order to make it easier to comply with Ansible guidelines.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

fortios_web_proxy_profile

##### ANSIBLE VERSION
```
ansible 2.8.0.dev0 (new_module ddbbe5dfa5) last updated 2018/09/24 14:54:57 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/magonzalez/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/magonzalez/ansible/lib/ansible
  executable location = /home/magonzalez/ansible/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

### CODE REVIEW

**Comment by thomnico** [comment]
shipit

**Review by NilashishC — CHANGES_REQUESTED** [review]
@mamunozgonzalez Please make the changes as suggested in #52851 in this PR as well.

**Comment by mamunozgonzalez** [comment]
I have updated it according to the comments we discussed. Thanks!

**Comment by thomnico** [comment]
shipit

File: lib/ansible/modules/network/fortios/fortios_web_proxy_profile.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        "username": {"required": True, "type": "str"},
+        "password": {"required": False, "type": "str", "no_log": True},
+        "vdom": {"required": False, "type": "str", "default": "root"},
+        "https": {"required": False, "type": "bool", "default": True},
+        "web_proxy_profile": {
+            "required": False, "type": "dict",
+            "options": {
+                "state": {"required": True, "type": "str",
+                          "choices": ["present", "absent"]},
+                "header-client-ip": {"required": False, "type": "str",
```
Comment (sivel) [inline lib/ansible/modules/network/fortios/fortios_web_proxy_profile.py:332]: For what it's worth, these names are technically invalid.  We are working on adding CI checks to ensure that arguments are valid python identifiers (variable names), which prohibits using hyphens in an argument name.

At the moment the tool we are using in CI doesn't recurse suboptions, only the top level, but will start validating the names of suboptions soon, at which point, these argument names will be classified as invalid.

Is there the potential that these could be converted now?

Comment (thomnico) [reply lib/ansible/modules/network/fortios/fortios_web_proxy_profile.py:332]: Changing rules 2 weeks before a release is a sure method to loose contributors.

We agreed to fix those with some reasonable time to comply.  Can we log a bug and fix for 2.9 ?

In addition it is basically forcing us a rule on the naming of our own API.

**Comment by NilashishC** [comment]
@thomnico @mamunozgonzalez As discussed, we're merging these PRs for 2.8 release. However, we would really like to have these modules patched with the required fixes in the future. Thank you.

**Pull request merged**